dbms/src/Core/SettingsCommon.h

```diff
@@ -76,6 +77,7 @@ struct SettingMaxThreads
     UInt64 value;
     bool is_auto;
     bool changed = false;
+    bool immutable = false;
```

Comment:
Probably `readonly` is a better name because everyone knows what `readonly` means.